physionet-django/console/utility.py

```diff
@@ -171,3 +177,120 @@ def paginate(request, to_paginate, maximun):
     paginator = Paginator(to_paginate, maximun)
     paginated = paginator.get_page(page)
     return paginated
+
+
+def create_doa_draft(project):
+    """
+    Create a draft DOI with some basic information about the project.
+
+    A POST request is done to set the base information.
+    The assigned DOI is returned to be used in the template.
+    """
+    url = settings.DATACITE_TEST_URL
+    if url == False:
+        return ""
+
+    prefix = settings.DATACITE_TEST_PREFIX
+    username = settings.DATACITE_TEST_USER
+    password = settings.DATACITE_TEST_PASS
+    current_site = Site.objects.get_current()
+    if settings.PRODUCTION:
+        prefix = settings.DATACITE_PREFIX
+        url = settings.DATACITE_URL
+        username = settings.DATACITE_USER
+        password = settings.DATACITE_PASS
+
+    headers = {'Content-Type': 'application/vnd.api+json'}
+
+    Dataset = ['Database', 'Challenge', 'Model']
+    Software = ['Software']
+
+    if project.resource_type.name in Dataset:
+        resource_type = 'Dataset'
+    elif project.resource_type.name in Software:
+        resource_type = 'Software'
+
+    author_list = project.author_list().order_by('display_order')
+    authors = []
+    for author in author_list:
+        authors.append({"name": author.get_full_name()})
+
+    description = strip_tags(project.abstract)
+
+    payload = {
+      "data": {
+        "type": "dois",
+        "attributes": {
+          "event": "draft",
+          "prefix": prefix,
+          "creators": authors,
+          "titles": [{
+            "title": project.title
+          }],
+          "publisher": current_site.name,
+          "publicationYear": timezone.now().year,
+
+          "types": {
+            "resourceTypeGeneral": resource_type
+          },
+          "url": 'https://{}'.format(current_site.domain),
+          "descriptions": [
+          {
+            "description": description,
```

Comment:
As we are assigning each version a new DOI, we should populate the version property (ID 15) in the DataCite metadata. See page 26 of https://schema.datacite.org/meta/kernel-4.3/doc/DataCite-MetadataKernel_v4.3.pdf

> Suggested practice: track
major_version.minor_version.
Register a new identifier for a
major version change.
Individual stewards need to
determine which are major
vs. minor versions27
.
Software engineering practice
follows this approach of
tracking changes and giving
new version numbers.
May be used in conjunction
with properties 11 and 12

Reply:
I can set the version, but for some ODD reason it doesn't show in the test site once I add it.

Also, there are 3 variables that could be used for version:
 - version
 - metadataVersion
 - schemaVersion

I added all 3, and only "schemaVersion" is shown.
I can keep them all if you want it.

Reply:
As an example of the DataCite JSON output of a DOI...
```
{
  "id": "https://doi.org/XX.XXXX/XXX-XXX",
  "doi": "XX.XXXX/XXX-XXX",
  "url": "https://localhost:8000",
  "types": {
    "resourceTypeGeneral": "Dataset"
  },
  "creators": [
    {
      "name": "PERSON LASTNAME"
    }
  ],
  "titles": [
    {
      "title": "project.title"
    }
  ],
  "publisher": "PhysioNet-dev",
  "publicationYear": 2020,
  "descriptions": [
    {
      "description": "description",
      "descriptionType": "Abstract"
    }
  ],
  "schemaVersion": "2.3",
  "providerId": "physiont",
  "clientId": "physiont.physiont",
  "agency": "DataCite",
  "state": "draft"
}
```

Reply:
Since I am using the API reference only the things here can be sent. 
https://support.datacite.org/reference/dois-2#post_dois

ALSO, there is NO version in the online form to register a DOI.

Reply:
Just in case, only set the following:
```
          "version": project.version,
          "metadataVersion": project.version,
          "schemaVersion": project.version,
```

Reply:
`schemaVersion` and `metadataVersion` refer to the version of the metadata schema, so are not relevant here and can be left empty. The field that we would like to populate is `version` (which has an ID of 15 in the schema).

Reply:
So the example above is the actual output if I set
```
          "version": project.version,
          "metadataVersion": project.version,
          "schemaVersion": project.version,
```

If I only set the version, I see NO version addition to the DOI. It's sent, but ignored by the API apparently.

Reply:
If you log into https://doi.test.datacite.org/, and check the draft DOI that ends with "g1hr-e130" you will see the word "Version" but its empty. 
That DOI was created with:
```
          "version": project.version,
```